File: 2496-maximum-value-of-a-string-in-an-array/2496-maximum-value-of-a-string-in-an-array.go
func maximumValue(strs []string) int {
	answer := 0
	for _, str := range strs {
		value := getLength(str)
		if value > answer {
			answer = value
		}
	}
	return answer
}

func getLength(str string) int {
	if isNumeric(str) {
		num, _ := strconv.ParseInt(str, 10, 64)
		return int(num)
	} else {
		return len(str)
	}
}

func isNumeric(str string) bool {
	for _, char := range str {
		if char < '0' || char > '9' {
			return false
		}
	}
	return true
}

File: 2496-maximum-value-of-a-string-in-an-array/2496-maximum-value-of-a-string-in-an-array.java
class Solution {
    public int maximumValue(String[] strs) {
        int answer = 0;
        for (String str: strs) {
            int value = getLength(str);
            if (value > answer) {
                answer = value;
            }
        }
        return answer;
    }

    public int getLength(String str) {
        if (isNumeric(str)) {
            return Integer.parseInt(str);
        } else {
            return str.length();
        }
    }

    public boolean isNumeric(String str) {
        for (int i =0; i <str.length(); i++) {
            if (str.charAt(i) < '0' || str.charAt(i) > '9') {
                return false;
            }
        }
        return true;
    }
}